include/swift/AST/ASTMangler.h

```diff
@@ -151,7 +151,17 @@ class ASTMangler : public Mangler {
                                              Type FromType, Type ToType,
                                              Type SelfType,
                                              ModuleDecl *Module);
-  
+
+  // SWIFT_ENABLE_TENSORFLOW
+  // Mangle the autodiff associated function with the given:
+  // - Mangled original function name.
+  // - Associated function kind.
+  // - Parameter/result indices.
+  // - Flag representing whether the function is a differential/pullback.
+  std::string mangleAutoDiffAssociatedFunctionHelper(
```

Comment:
Associated function kind is orthogonal to linear maps. I’d suggest adding to separate mangling helpers, one for associates functions and one for linear maps. I think you should also add a LinearMapKind enum to distinguish between differential and pullback.

Reply:
Added `AutoDiffLinearMapKind` in https://github.com/apple/swift/pull/26389/commits/221ddb0d76529c6f043b98af2e3734bdb5682be9. Let me know if you prefer the name `LinearMapKind`.

Added `ASTMangler::mangleAutoDiffLinearMapHelper` in https://github.com/apple/swift/pull/26389/commits/c5b2018a91fe40a34c44d4e7a164fc96f19d34b9.